src/mount/conf.cc

```diff
@@ -66,24 +65,21 @@ extern "C" void mount_ceph_get_config_info(const char *config_file,
 	continue;
     }
 
-    std::string addr;
-    addr += eaddr.ip_only_to_str();
-    addr += ":";
-    addr += std::to_string(eaddr.get_port());
-    /* If this will overrun cci_mons, stop here */
-    if (monaddrs.length() + 1 + addr.length() + 1 > sizeof(cci->cci_mons))
+    /*
+     * If this will overrun cci_mons, stop here. one +1 is for the comma and
+     * the other is for the null terminator.
+     */
+    if ((unsigned)oss.tellp() + 1 + eaddr.get_sockaddr_len() + 1 > sizeof(cci->cci_mons))
```

Comment:
This is wrong.  `get_sockaddr_len()` has nothing to do with the string representation generated by `<<`.

I would actually question the "stop here" logic because it can silently truncate the list of monitors.  I think we should get rid of this check here and perform it only before calling `strcpy()`.  If the entire `oss` buffer doesn't fit, print an error and leave `cci->cci_mons` empty.